src/Core/Maintenance/Test/System/Struct/DatabaseConnectionInformationTest.php

```diff
@@ -30,4 +30,16 @@ public function testItEncodesSpecialCharsInDbPasswords(): void
 
         static::assertEquals('mysql://user:ultra%3Fsecure%23@mysql:3306/test_db', $dbConnectionInformation->asDsn());
     }
+
+    /**
+     * @backupGlobals enabled
+     */
+    public function testItAllowsAnEmptyDbPassword(): void
+    {
+        unset($_SERVER['DATABASE_URL'], $_ENV['DATABASE_URL']);
```

Comment:
please use the `\Shopware\Core\Framework\Test\TestCaseBase\EnvTestBehaviour` trait and then use 
```php
$this->setEnvVars(['DATABASE_URL' => 'mysql://user:@mysql:3306/test_db']);
```

that way you can also remove the `backupGlobals` annotation

please also update the first testcase

Reply:
Switched to use trait for both tests.